### PR TITLE
Add support for taxes on expenses

### PR DESCRIPTION
**Pull request by Betree**
Part of https://github.com/opencollective/opencollective/issues/4532
Requirement for https://github.com/opencollective/opencollective-frontend/pull/7605

- [x] Add support in mutation when creating
- [x] Add support in mutation when editing
- [x] Record things properly in the ledger
- [x] Add a test for creating an expense with VAT
- [x] Add a test for editing an expense with VAT
- [x] Add a test for paying an expense with VAT

### CODE REVIEW

**Review by kewitz — COMMENTED** [review]
Some notes I had when getting context to review the frontend.

File: server/graphql/v2/input/ExpenseUpdateInput.ts [hunk 1]
```diff
@@ -72,5 +73,9 @@ export const ExpenseUpdateInput = new GraphQLInputObjectType({
       type: LocationInput,
       description: 'The address of the payee',
     },
+    tax: {
```
[the remark this answers is not in the document]
Comment (Betree) [reply server/graphql/v2/input/ExpenseUpdateInput.ts:76]: We actually defined that singular should be the default: https://github.com/opencollective/opencollective/issues/4378

File: server/graphql/common/expenses.ts [hunk 2]
```diff
@@ -708,9 +710,30 @@ type ExpenseData = {
...
   description?: string;
   currency?: string;
+  tax?: TaxDefinition[];
+};
+
+const checkTaxes = (account, host, expenseType: string, taxes): void => {
+  if (!taxes?.length) {
+    return;
+  } else if (taxes.length > 1) {
+    throw new ValidationFailed('Only one tax is allowed per expense');
```
[the remark this answers is not in the document]
Comment (Betree) [reply server/graphql/common/expenses.ts:723]: For this MVP we only support a single tax on expenses, that's what this line enforces. But in theory, multiple taxes can apply to the same transaction. For example, in France, your invoice can have VAT (value-added tax) + "recyclability" tax.

This is consistent with what we do for orders.

File: server/graphql/v2/interface/Transaction.js [hunk 3]
```diff
@@ -332,10 +331,16 @@ export const TransactionFields = () => {
...
+        const tax = transaction.data?.tax;
+        if (!tax) {
           return null;
         } else {
-          return pick(transaction.data.tax, ['id', 'percentage']);
+          return {
+            id: tax.id,
+            type: tax.id,
+            percentage: Math.round(tax.percentage), // Does not support float
+            rate: tax.percentage,
```
[the remark this answers is not in the document]
Comment (Betree) [reply server/graphql/v2/interface/Transaction.js:342]: That's a good idea, I'll look into that

**Pull request merged**